code/parse/sexp.cpp

```diff
@@ -22970,11 +22970,11 @@ bool is_sexp_true(int cur_node, int referenced_node)
 {
 	int result = eval_sexp(cur_node, referenced_node);
 
-	// any SEXP_KNOWN_TRUE result will return SEXP_TRUE from eval_sexp, but let's be defensive
+	// any* SEXP_KNOWN_TRUE result will return SEXP_TRUE from eval_sexp, but let's be defensive
+	// *for almost all return paths... there is one path where it has not yet been proven that it won't
```

Comment:
...Well, what is that code path, and why not just guarantee it instead of doing a partial revert of PR #42 "just in case"?

Reply:
Oh dear, it looks like PR #42 may need more scrutinizing if we're adding KNOWN_ values back in. :-/

I made these changes because of lines 25573 et seq:
```
		if ( Sexp_nodes[cur_node].value == SEXP_NAN ) {	// if we had a nan, but now don't, reset the value
			Sexp_nodes[cur_node].value = SEXP_UNKNOWN;
			return sexp_val;
		}
```

...Having gone back and looked at those lines, though, I see that the cases where `sexp_val` is KNOWN_ or FOREVER_ are *already* checked in the several if() blocks proceeding that one.  So it turns out that the leaky code path is, in fact, not leaky at all.  I'll revert those modifications.